opal/datatype/opal_datatype_copy.c

```diff
@@ -76,7 +77,7 @@ static void *opal_datatype_accelerator_memcpy(void *dest, const void *src, size_
     else if (0 >= dst_type && 0 < src_type) {
 	copy_type = MCA_ACCELERATOR_TRANSFER_DTOH;
     }
-    else if (0 < dst_type && 0 >= dst_type) {
+    else if (0 < dst_type && 0 >= src_type) {
```

Comment:
😨

Reply:
Yeah, this was a compiler warning that turned up a legit bug.